sanity-io/structured-content-2022
# On-demand fetch Venues for `venuesSection`, Sponsors for `sponsorsSection`

# On-demand fetch Venues for `venuesSection`, Sponsors for `sponsorsSection`

## Intent

When rendering a `venuesSection` or `sponsorsSection`, fetch the associated entities and pass them down through `<TextBlock />`. This is an improvement over the previous version which would fetch the entities for each page load.

## Testing this PR

1. Open [/](https://structured-content-2022-web-c6a7b66hq.sanity.build/) 
2. Verify that `<VenueNames />` is still correctly rendered, although it is fetched as part of a Landing Page now and no longer hardcoded in _[[...slug]].tsx_
3. Verify that Sponsors are correctly rendered now
